bluefog/torch/optimizers.py

```diff
@@ -756,6 +871,31 @@ def DistributedNeighborAllreduceOptimizer(optimizer, model):
     Arguments:
         optimizer: Optimizer to use for computing gradients and applying updates.
         model: The model or a list of models you want to train with.
+        num_steps_per_communication: Number of expected model forward function calls before each
+                                     communication. This allows local model parameter updates
+                                     per num_steps_per_communication before reducing them over
+                                     distributed computation resources.
+
+    Example for  two scenarios to use num_steps_per_communication:
+        Scenario 1) w_{i+1} = neighbor_allredce(w_i) + \sum \grad_j(w_i)
+                    No local model parameter update, accumulate the gradients for reducing
```

Comment:
Maybe just use the wording should be enough.

Scenario 1) Local accumulation of gradient without update model. (Used in large batch size or large model cases)
Scenario 2) Local updating the model. (Used in case that decreasing the communication).